kong/db/schema/others/declarative_config.lua

```diff
@@ -392,55 +404,37 @@ local function generate_uuid(namespace, name)
 end
 
 
-local function generate_ids(input, known_entities, parent_entity)
-  for _, entity in ipairs(known_entities) do
-    if type(input[entity]) ~= "table" then
-      goto continue
-    end
+local function generate_ids(input, known_entities)
+  traverse_nested(input, known_entities, "pre", nil,
+                  function(entity, item, _, parent_fk, child_key)
 
-    local parent_fk
-    local child_key
-    if parent_entity then
-      local parent_schema = all_schemas[parent_entity]
-      if parent_schema.fields[entity] then
-        goto continue
-      end
-      parent_fk = parent_schema:extract_pk_values(input)
-      child_key = foreign_children[parent_entity][entity]
-    end
+    local schema = all_schemas[entity]
```

Comment:
I chose not to do this because this would make the new parameter dependent on the first (given an `entity` string, there would only ever be one correct value for `schema`, so this is not really a "parameter" as in "something you can tweak"). The fact that both uses of `traverse_nested` use that same value is an implementation coincidence of the functions, not a property of the `traverse_nested` contract.